### PR TITLE
Openstack4J integration

### PR DESCRIPTION
**Pull request by pawelpros**
Removed support for jcloud and updated code to use openstack4j library with keystone version 3.

Additionally:

1. Updated floating ip behavior
2. updated deprecated usage of extensions to proper supported ones
3. updated OSGI
4. removed all dependencies connected with jcloud
5. added dependencies to openstack4j

### CODE REVIEW

File: src/main/java/org/osc/controller/nsc/api/openstack4j/BaseOpenstack4jApi.java [hunk 1]
```diff
@@ -14,37 +14,32 @@
...
+    private Endpoint endPoint;
+    private KeystoneProvider keystoneProvider;
 
-    public BaseJCloudApi(Endpoint endPoint) {
+    protected BaseOpenstack4jApi(Endpoint endPoint) {
         this.endPoint = endPoint;
+        this.keystoneProvider = KeystoneProvider.getInstance();
+    }
+
+    protected OSClient.OSClientV3 getOs() {
```
Comment (emanoelxavier) [inline src/main/java/org/osc/controller/nsc/api/openstack4j/BaseOpenstack4jApi.java]: getOSTClient or getOpenStackClient

File: src/test/java/org/osc/controller/nsc/OSGiIntegrationTest.java [hunk 2]
```diff
@@ -42,8 +43,15 @@
 import org.osgi.framework.ServiceObjects;
 import org.osgi.util.tracker.ServiceTracker;
 
-@RunWith(PaxExam.class)
-@ExamReactorStrategy(PerMethod.class)
+/*TODO: test is commented, because there is a problem with openstack4j dependecies injected directly into osgi (whicch is build during that test)
```
Comment (emanoelxavier) [inline src/test/java/org/osc/controller/nsc/OSGiIntegrationTest.java]: Is this due to an introduced bug or a limitation/issue on openstack4j? When are you planning to re-enable?

Comment (pawelpros) [inline src/test/java/org/osc/controller/nsc/OSGiIntegrationTest.java]: This is rather missed configuration as we are not good at OSGI - needed few days to investigate missing dependencies to enable test but we didn't have time to finish it till our transfer to new projects. This topic was passed to paremus

Comment (pawelpros) [inline src/test/java/org/osc/controller/nsc/OSGiIntegrationTest.java]: This is rather missed configuration as we are not good at OSGI - needed few days to investigate missing dependencies to enable test but we didn't have time to finish it till our transfer to new projects. This topic was passed to paremus

**Comment by emanoelxavier** [comment]
@arvindn05 Is this still a valid PR?

**Comment by arvindn05** [comment]
@dmitryintel was going to update to see if openstack4j works as is. If it works as is(except for the unit tests) we can checkin. 

If not we will can leave jclouds as is.

But given that NSC is deprecated and we are planning to move to stub approach for this plugin....maybe we just drop calling openstack API's altogether.

**Comment by arvindn05** [comment]
nsc plugin is going to use a dummy datatbase to store the API calls and results. will no longer need to call openstack API's

**Pull request closed without merging**